Classes/Plugin/OaiPmh.php

```diff
@@ -449,35 +456,42 @@ protected function resume()
      */
     protected function verbGetRecord()
     {
-        if (count($this->piVars) != 3 || empty($this->piVars['metadataPrefix']) || empty($this->piVars['identifier'])) {
+        if (count($this->piVars) !== 3 || empty($this->piVars['metadataPrefix']) || empty($this->piVars['identifier'])) {
             return $this->error('badArgument');
         }
-        if (!in_array($this->piVars['metadataPrefix'], array_keys($this->formats))) {
+        if (!array_key_exists($this->piVars['metadataPrefix'], $this->formats)) {
             return $this->error('cannotDisseminateFormat');
         }
         $where = '';
         if (!$this->conf['show_userdefined']) {
-            $where .= ' AND tx_dlf_collections.fe_cruser_id=0';
+            $where .= 'AND tx_dlf_collections.fe_cruser_id=0 ';
         }
-        $record = $GLOBALS['TYPO3_DB']->exec_SELECT_mm_query(
-            'tx_dlf_documents.*,GROUP_CONCAT(DISTINCT tx_dlf_collections.oai_name ORDER BY tx_dlf_collections.oai_name SEPARATOR " ") AS collections',
-            'tx_dlf_documents',
-            'tx_dlf_relations',
-            'tx_dlf_collections',
-            'AND tx_dlf_documents.record_id=' . $GLOBALS['TYPO3_DB']->fullQuoteStr($this->piVars['identifier'], 'tx_dlf_documents')
-                . ' AND tx_dlf_documents.pid=' . intval($this->conf['pages'])
-                . ' AND tx_dlf_collections.pid=' . intval($this->conf['pages'])
-                . ' AND tx_dlf_relations.ident=' . $GLOBALS['TYPO3_DB']->fullQuoteStr('docs_colls', 'tx_dlf_relations')
-                . $where
-                . Helper::whereClause('tx_dlf_collections'),
-            '',
-            '',
-            '1'
-        );
-        if (!$GLOBALS['TYPO3_DB']->sql_num_rows($record)) {
+
+        $connection = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionForTable('tx_dlf_documents');
+
+        $sql = 'SELECT `tx_dlf_documents`.*, GROUP_CONCAT(DISTINCT `tx_dlf_collections`.oai_name ORDER BY tx_dlf_collections.oai_name SEPARATOR " ") AS `collections` ' .
```

Comment:
Please consider using table and field enclosing consistently. Currently some table names are enclosed in backticks and some are not. Generally we should enclose all table and field names in backticks to be safe.

Reply:
done